packages/block-library/src/embed/edit.native.js

```diff
@@ -282,11 +289,13 @@ const EmbedEdit = ( props ) => {
 			) }
 			<EmbedBottomSheet
 				value={ url }
-				isVisible={ isEditingURL }
-				onClose={ () => setIsEditingURL( false ) }
+				isVisible={ showEmbedBottomSheet }
+				onClose={ () => {
+					setShowEmbedBottomSheet( false );
+				} }
 				onSubmit={ ( value ) => {
-					setIsEditingURL( false );
 					setAttributes( { url: value } );
+					setIsEditingURL( false );
```

Comment:
I found out that the order here is important, we need to update the URL attribute before the following side-effect is triggered when the `isEditingURL` value is changed:
https://github.com/WordPress/gutenberg/blob/6dd74384f726641d2bd40b5d58a3c9f477a1daa0/packages/block-library/src/embed/edit.native.js#L171-L194

Reply:
Do you think maybe we should add a comment here as well?

Reply:
Good point, I'll add a comment here to clarify it 👍 .

Reply:
I added your suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/35013/commits/9bbcc377c711d32f49dcb6b3a7d44fa789274c8a).